scanner/decisions/0005-scannerv4-read-cvss-data.md

```diff
@@ -0,0 +1,45 @@
+
+The CVSS Updater will attempt to download each zip file from the Google bucket up to five times. If all five attempts fail, it will skip that particular file. For instance
+
+| Action                                                   | Result                                         | Next Step                               |
+|----------------------------------------------------------|------------------------------------------------|-----------------------------------------|
+| GET /bucket/1st-severity.zip                            | Success                                        | download to a tmp directory (disk)      |
+| GET /bucket/2nd-severity.zip                             | Success                                        | download to same tmp directory (disk)   |
+| GET /bucket/3rd-severity.zip                             | Failed (e.g. 401, 403, 505, Timeout)           | Retry 5 times and SKIP                  |
+| Create a final zip file from all jsons in downloaded zip files and save to store.Snapshot      | Success                                        | Wait 4 hours for another update         |
+
+
+## Consequences
+
+One drawback of the NVD CVE API is its rate limit: only 5 requests are allowed per 30 seconds without an API Key. This restriction necessitates that our workflow fetch data at a slower pace. Additionally, transitioning from the NVD JSON feed to the NVD CVE API brings in pagination limitations. For example, when fetching data with a "low" severity, the API provides a maximum of 2000 CVEs per response. This means multiple requests are required to retrieve the complete dataset for "low" severity. The result is an increased number of smaller JSON files. Consequently, we need to compress all JSON files for a particular severity into a single zip archive.
+
+Changing from the NVD JSON feed to the NVD CVE API also give us the conveniences of no need to filter out data with out valid CVSS V3 metrics, as mentioned above. 
+
+Following this modification, Central will no longer have a dependency on Claircore. The Claircore enricher will be used in Scanner V4, which adopts Claircore as its primary scanning engine by design.
+
+Per the NVD's documentation (https://nvd.nist.gov/General/News/change-timeline), the NVD CVE API omits CVSS v3 vector strings with a 'NONE' severity. This ensures we only obtain data bearing valid CVSS V3 metrics.
+
+If there are discrepancies in the CVSS data during fetching or downloading, the index report will show an incomplete CVSS score, and certain CVEs might be without their respective scores. Because UpdateEnrichments in Claircore initiates a new UpdateOperation, adds the provided EnrichmentRecord(s), and ensures that enrichments from past updates are not accessible to clients.
```

Comment:
Do you have any recommendations regarding this matter? Is it acceptable for customers to have missing CVSS scores in their scanning reports for a few hours? @RTann 

cc: @jvdm

Reply:
I think it is very possible we will be missing CVSS scores for vulnerabilities, as it may take NVD time to add the vulnerability into its database. This is something we deal with now, but we opt to block those vulns until CVSS scores are present (or we manually add the CVEs ourselves).

I think it's best to allow this scenario, and call CVEs without CVSS scores valid. No reason a user should not know about a potential vulnerability just because it's missing a CVSS score. The vendor may have assigned a severity which should help the user assess the risk

Reply:
@ross The scenario Yi is calling out is when we cannot pull CVSS scores for a given "group" (in his proposal, severities). He's proposing we attempt to update the other successful severities, which means we will effectively delete the missing ones since ClairCore handles these updates as if they were "Snapshots" rather than insertions/append.

Reply:
```suggestion
When some parts of the CVSS data cannot be accessed or downloaded during the process of fetching or downloading, the vulnerability report may show missing CVSS scores for some vulns that had them before. This happens because the `UpdateEnrichments` feature in ClairCore adds the EnrichmentRecord(s) provided, but it ensures that the enrichments from previous updates cannot be accessed by clients, effectively removing them from vulnerability reports.```

Reply:
You've got the wrong @ross